packages/providers/test/provider.test.ts

```diff
@@ -281,29 +296,30 @@ describe('Provider', () => {
     expect(producedBlocks).toEqual(expectedBlocks);
   });
 
-  it('can cacheUtxo [undefined]', () => {
-    const provider = new Provider('http://127.0.0.1:4000/graphql');
+  it('can cacheUtxo [undefined]', async () => {
+    const provider = await Provider.connect('http://127.0.0.1:4000/graphql');
 
     expect(provider.cache).toEqual(undefined);
   });
 
-  it('can cacheUtxo [numerical]', () => {
-    const provider = new Provider('http://127.0.0.1:4000/graphql', {
+  it('can cacheUtxo [numerical]', async () => {
+    const provider = await Provider.connect('http://127.0.0.1:4000/graphql', {
       cacheUtxo: 2500,
     });
 
     expect(provider.cache).toBeTruthy();
     expect(provider.cache?.ttl).toEqual(2_500);
   });
 
-  it('can cacheUtxo [invalid numerical]', () => {
-    expect(() => new Provider('http://127.0.0.1:4000/graphql', { cacheUtxo: -500 })).toThrow(
-      'Invalid TTL: -500. Use a value greater than zero.'
+  it('can cacheUtxo [invalid numerical]', async () => {
+    const { error } = await safeExec(() =>
+      Provider.connect('http://127.0.0.1:4000/graphql', { cacheUtxo: -500 })
     );
+    expect(error?.message).toMatch(/Invalid TTL: -500\. Use a value greater than zero/);
```

Comment:
@arboleya makes sense, but I don't think this was added in this PR. It just shows up in the changed files because I had to change that test to accomodate the new Provider API.

Reply:
That's fine. We can do another check as we go.